packages/itwin/tree-widget/src/tree-widget-react/components/trees/models-tree/internal/ModelsTreeVisibilityHandler.ts

```diff
@@ -430,25 +513,47 @@ class ModelsTreeVisibilityHandlerImpl implements HierarchyVisibilityHandler {
     ignoreTooltip,
     ...props
   }: GetGeometricElementVisibilityStatusProps & { ignoreTooltip?: boolean }): Observable<VisibilityStatus> {
-    const result = defer(() => {
+    const result: Observable<VisibilityStatus> = defer(() => {
       const viewport = this._props.viewport;
       const { elementId, modelId, categoryId } = props;
 
       if (!viewport.view.viewsModel(modelId)) {
         return of(createVisibilityStatus("hidden", getTooltipOptions("modelsTree.element.hiddenThroughModel", ignoreTooltip)));
       }
 
-      let status = this.getElementOverriddenVisibility(elementId, ignoreTooltip);
-      if (status) {
-        return of(status);
+      const elementStatus = this.getElementOverriddenVisibility(elementId, ignoreTooltip);
+      if (elementStatus?.state === "hidden") {
+        return of(elementStatus);
       }
```

Comment:
Is this correct now? An element could be hidden through its model selector or never display override, but if sub-model is not also hidden, then the end result should be "partial"